service/cmdcache/options.go

```diff
@@ -0,0 +1,12 @@
+package cmdcache
+
+type Option func(*Cache)
+
+func WithBatching(batchSize uint32) Option {
+	if batchSize == 0 {
+		panic("batch size must be higher than zero")
```

Comment:
greater than.

Do we want to panic if batch size is 1? We should probably not use `WithBatching(1)`, but there could be an argument made for allowing it to avoid having to remove the entire option when using the code in a loop with `WithBatching(k)`...